src/Engine/Objects/Player.cpp

```diff
@@ -4281,12 +4281,12 @@ bool Player::CompareVariable(VariableType VarNum, int pValue) {
     int actStat;                           // ebx@161
     int baseStat;                          // eax@161
 
-    if ((signed int)VarNum >= VAR_MapPersistentVariable_0 && VarNum <= VAR_MapPersistentVariable_74)
+    if (VarNum >= VAR_MapPersistentVariable_0 && VarNum <= VAR_MapPersistentVariable_74)
         return (uint8_t)stru_5E4C90_MapPersistVars.field_0[VarNum - VAR_MapPersistentVariable_0] >= pValue;  // originally (uint8_t)byte_5E4C15[VarNum];
 
     // not really sure whether the number gets up to 99, but can't ignore the possibility
-    if ((signed int)VarNum >= VAR_MapPersistentVariable_75 && VarNum <= VAR_MapPersistentVariable_99)
-        return (uint8_t)stru_5E4C90_MapPersistVars._decor_events[VarNum - VAR_MapPersistentVariable_75] >= pValue;
+    if (VarNum >= VAR_MapPersistentDecorVariable_0 && VarNum <= VAR_MapPersistentDecorVariable_24)
+        return (uint8_t)stru_5E4C90_MapPersistVars._decor_events[VarNum - VAR_MapPersistentDecorVariable_0] >= pValue;
```

Comment:
Sorry, i don't mean to be annoying, but given what i see:
1. stru123  -> Script/EventVariables
2. stru_5E4C90_MapPersistVars - > g_variables (or however we call global vars now)
3. stru123::field_0  -> stru123::mapVars ?
4. stru123::_decor_events-> stru123::decorVars ?
5. Given that you do (VarNum-VAR_MapPersistentDecorVariable_0) _decor_events size should be 25? please check if it's memset'd anywhere. i would also do it as a metod e.g SetDecorVar(VariableType slot, val) { assert var in (VAR_MapPersistentDecorVariable_0, VAR_MapPersistentDecorVariable_25);      this._decor_events[slot - VAR_MapPersistentDecorVariable_0] = val; }